tests/python/test_lightgbm_integration.py

```diff
@@ -237,7 +237,7 @@ def test_lightgbm_sparse_ranking_model(tmpdir):
 
     lgb_model_path = pathlib.Path(tmpdir) / "sparse_ranking_lightgbm.txt"
 
-    dtrain = lgb.Dataset(X, label=y, group=[X.shape[0]])
+    dtrain = lgb.Dataset(X, label=y, group=np.array([X.shape[0]], dtype=np.int32))
```

Comment:
This one is `lightgbm`'s responsibility to fix.

Passing a list for `group` ends up invoking `np.array(..., copy=False)`.

```text
    def _list_to_1d_numpy(
        data: Any,
        dtype: "np.typing.DTypeLike",
        name: str
    ) -> np.ndarray:
        """Convert data to numpy 1-D array."""
        if _is_numpy_1d_array(data):
            return _cast_numpy_array_to_dtype(data, dtype)
        elif _is_numpy_column_array(data):
            _log_warning('Converting column-vector to 1d array')
            array = data.ravel()
            return _cast_numpy_array_to_dtype(array, dtype)
        elif _is_1d_list(data):
>           return np.array(data, dtype=dtype, copy=False)
E           ValueError: Unable to avoid copy while creating an array as requested.
E           If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
E           For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.
```

Passing a `numpy` array instead avoids that code path and allows this test to pass on NumPy 2.x

Reply:
I didn't post enough stacktrace there.

To be clear, that is code inside `lightgbm`: https://github.com/microsoft/LightGBM/blob/e0ac63568c900405e5e438d95c539f3490919aab/python-package/lightgbm/basic.py#L358

Reply:
The `int32` is a bit unfortunate, but it, or `intc` is necessary here and seems totally safe to keep indefinitely.